source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -674,13 +675,21 @@ The event API here is assumed to be like the standard `Python Event
             wait()
 
     def setUpConnection():
+        # Take the mutex to avoid a data race becauase this code writes to the connection field and a concurrent
+        # cancelCheck call could be reading from it.
+        mutex.lock()
```

Comment:
In SPEC-1690 I've also introduced a lock. Can we use it the same in the pseudocode? IE rename it from "mutex" to "lock" and use a with-statement. Like this:
```python
with lock:
    connection = new Connection(serverAddress)
    set connection timeout to connectTimeoutMS
# Do any potentially blocking operations after releasing the mutex.
create the socket and perform connection handshake
```

Reply:
Thanks! I was just guessing on reasonable lock syntax. I pushed a new commit to match the pseudocode syntax from SPEC-1690.